packages/twenty-server/src/workspace/messaging/services/save-messages-and-create-contacts.service.ts

```diff
@@ -82,10 +84,15 @@ export class SaveMessagesAndCreateContactsService {
     if (isContactAutoCreationEnabled) {
       startTime = Date.now();
 
-      await this.createCompaniesAndContactsService.createCompaniesAndContacts(
-        connectedAccount.handle,
-        contactsToCreate,
-        workspaceId,
+      await workspaceDataSource?.transaction(
+        async (transactionManager: EntityManager) => {
```

Comment:
I think we should isolate this part into a dedicated job or through a listener+job now that they don't run in the same transaction.
If the parent job is not idempotent, it will try to reimport the same messages because of a failure in the saveContacts.
If the parent job is idempotent (as it should be), it will not find new messages to save so no participants and after a retry it will succeed without importing the participants that failed to be imported

Reply:
Automatically creating contacts could be done in a 2nd round, this should not affect email sync and at that point we should have all the necessary info to do that without querying google (we have the messageParticipants table up to date) that's why I'm suggesting creating a new job that would deal with that or even a listener on "messageParticipant.created" (though you would have to emit it yourself since you are not going through the API) that will enqueue its own job